src/object.c

```diff
@@ -733,10 +733,12 @@ size_t objectComputeSize(robj *o, size_t sample_size) {
         if (o->encoding == OBJ_ENCODING_HT) {
             d = o->ptr;
             di = dictGetIterator(d);
-            asize = sizeof(*o)+sizeof(dict)+(sizeof(struct dictEntry*)*dictSlots(d));
+            int dicthtNum = dictIsRehashing(d) + 1;
```

Comment:
I think the existing computation of asize is correct. sizeof(dict) includes size of the 2 hash tables used by the dictionary. 
Please refer to https://github.com/antirez/redis/blob/unstable/src/dict.h#L76-L82

Same is the case for changes below.